src/core/sorted_map.cc

```diff
@@ -435,13 +435,11 @@ int SortedMap::DfImpl::Add(double score, sds ele, int in_flags, int* out_flags,
       return 1;
     }
 
-    auto [newk, added] = score_map->AddOrUpdate(string_view{ele, sdslen(ele)}, score);
-    DCHECK(added);
+    obj = score_map->AddUnique(string_view{ele, sdslen(ele)}, score);
 
     *out_flags = ZADD_OUT_ADDED;
     *newscore = score;
-    obj = newk;
-    added = score_tree->Insert(obj);
+    bool added = score_tree->Insert(obj);
```

Comment:
So that's the only external use so far? I also assume we can use this when we created the key for regular hashes (small nit)

Reply:
you can see that also reimplemented `StringMap::AddOrSkip` that also benefits from this function. 
I agree that `AddOrUpdate` functions should disappear from the interface because they will always be inefficient.

Reply:
hmm, actually it's not easy to remove `AddOrUpdate` because of technical reasons of how entries are created in DenseSet. Will keep thinking about it.